docs/data/stashcache/install-origin.md

```diff
@@ -71,16 +71,13 @@ AFTER the OSG configuration).
 
 You _must_ configure every variable in `/etc/xrootd/10-common-site-local.cfg` and `/etc/xrootd/10-origin-site-local.cfg`.
 
-The mandatory variables to configure in `10-common-site-local.cfg` are:
+The mandatory variables to configure are:
 
-* `set rootdir = /mnt/stash`: the mounted filesystem path to export.
-  This document refers to this path as `/mnt/stash`.
-* `set resourcename = YOUR_RESOURCE_NAME`: the resource name registered with OSG.
-
-The mandatory variable to configure in `10-origin-site-local.cfg` is:
-
-* `set originexport = /VO`: the directory relative to `rootdir` that is the top of the exported namespace
-  for the origin services
+| File                     | Config line                             | Description                                                                                           |
+|--------------------------|-----------------------------------------|-------------------------------------------------------------------------------------------------------|
+| 10-common-site-local.cfg | `set rootdir = /mnt/stash`              | The mounted filesystem path to export; this document calls it `/mnt/stash`                            |
+| 10-common-site-local.cfg | `set resourcename = YOUR_RESOURCE_NAME` | The resource name registered with OSG                                                                 |
```

Comment:
During the tests, I've figured `set resourcename = YOUR_RESOURCE_NAME` variable is not mandatory if you set (uncomment) `set DisableOsgMonitoring = 1`. I.e. if this is set, service will start:
```
...
#set resourcename = OSG_RESOURCE_NAME

# To opt out of OSG monitoring, uncomment the following line:
#
set DisableOsgMonitoring = 1
```
Should we make note of that in the documentation? Otherwise, if some of these mandatory variables are not set service start fails always with a not very meaningful message (it points to the different config which is then misleading for admin where to look:
```
190501 15:57:26 14268 XrdStream: Variable resourcename is undefined.
190501 15:57:26 14268 XrdConfig: sitename value not specified
Config continuing with file /etc/xrootd/config.d/40-osg-paths.cfg ...
=====> all.adminpath /var/spool/xrootd
Config continuing with file /etc/xrootd/config.d/50-stash-origin-authz.cfg ...
=====> xrd.allow host *
Config continuing with file /etc/xrootd/config.d/50-stash-origin-paths.cfg ...
Config continuing with file /etc/xrootd/config.d/90-xcache-logging.cfg ...
=====> xrd.trace all -sched
190501 15:57:26 14268 XrdConfig: Unable to read config file /etc/xrootd/xrootd-stash-origin.cfg; invalid argument
```
Still, it says which variable is undefined but doesn't really say in which file. But this is probably not for documentation rather to address in the code for the next release?

Reply:
1. I don't want to give people the guarantee that things will work if they don't set the resourcename. We might add additional config that depends on the resourcename later.

2. Yeah we could probably use a troubleshooting section with these common errors. It's xrootd that's doing the config parsing so any code changes would have to happen there...
Perhaps we could make an osg-configure-xcache to make some of this config easier...